module/zfs/arc.c

```diff
@@ -3472,7 +3472,8 @@ arc_prune_async(int64_t adjust)
 
 		refcount_add(&ap->p_refcnt, ap->p_pfunc);
 		ap->p_adjust = adjust;
-		taskq_dispatch(arc_prune_taskq, arc_prune_task, ap, TQ_SLEEP);
+		(void) taskq_dispatch(arc_prune_taskq, arc_prune_task,
+			ap, TQ_SLEEP);
```

Comment:
This is actually a real potential bug.  In the failure case where `taskq_dispatch()` fails we need to drop the `p_refcnt` which would normally have been run by `arc_prune_task` but now won't be.